### PR TITLE
Prefix metric namespaces with tempo instead of cortex

### PR DESCRIPTION
**Pull request by mapno**
**What this PR does**:

Changes metrics prefixed with `cortex_` to `tempo_`.

Affected metrics

```
    "cortex_cache_background_queue_length",
    "cortex_cache_dropped_background_writes_total",
    "cortex_distributor_ingester_clients",
    "cortex_dns_failures_total",
    "cortex_dns_lookups_total",
    "cortex_experimental_features_in_use_total",
    "cortex_kv_request_duration_seconds_bucket",
    "cortex_kv_request_duration_seconds_count",
    "cortex_kv_request_duration_seconds_sum",
    "cortex_member_consul_heartbeats_total",
    "cortex_memcache_client_servers",
    "cortex_memcache_client_set_skip_total",
    "cortex_memcache_request_duration_seconds_bucket",
    "cortex_memcache_request_duration_seconds_count",
    "cortex_memcache_request_duration_seconds_sum",
    "cortex_query_frontend_connected_clients",
    "cortex_query_frontend_discarded_requests_total",
    "cortex_query_frontend_queue_duration_seconds_bucket",
    "cortex_query_frontend_queue_duration_seconds_count",
    "cortex_query_frontend_queue_duration_seconds_sum",
    "cortex_query_frontend_queue_length",
    "cortex_ring_member_heartbeats_total",
    "cortex_ring_member_tokens_owned",
    "cortex_ring_member_tokens_to_own",
    "cortex_ring_members",
    "cortex_ring_oldest_member_timestamp",
    "cortex_ring_tokens_total",
    "cortex_shutdown_duration_seconds_bucket",
    "cortex_shutdown_duration_seconds_count",
    "cortex_shutdown_duration_seconds_sum"
```

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 * [ENHANCEMENT] Added the ability to multiple span metrics by an attribute such as `X-SampleRatio` [#2172](https://github.com/grafana/tempo/pull/2172) (@altanozlu)
 * [BUGFIX] Apply `rate()` to bytes/s panel in tenant's dashboard. [#2081](https://github.com/grafana/tempo/pull/2081) (@mapno)
 * [CHANGE] Update Go to 1.20 [#2079](https://github.com/grafana/tempo/pull/2079) (@scalalang2)
+* [CHANGE] Change metrics prefixed with `cortex_` to `tempo_` [#2204](https://github.com/grafana/tempo/pull/2204) (@mapno)
```
Comment (electron0zero) [inline CHANGELOG.md:30]: let's mark it as breaking change to inform community to update thair dashbnoards and alert rules.

Comment (electron0zero) [reply CHANGELOG.md:30]: also let's update dashboards and alert rules in `tempo-mixin`

Comment (mapno) [reply CHANGELOG.md]: Done!

**Comment by vainikkaj** [comment]
I recently tried to review existing metrics and I totally missed most of these because of their naming 😂

**Review by electron0zero — APPROVED** [review]

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**